plugins/vpc-bridge/config/k8s/pipe.go

```diff
@@ -0,0 +1,183 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+//go:build enablek8sconnector && windows
+
+package k8s
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"time"
+
+	winio "github.com/Microsoft/go-winio"
+	log "github.com/cihub/seelog"
+	"golang.org/x/sys/windows"
+)
+
+const (
+	//PipeConnectionTimeout Represents timeout for reading from connection object.
+	PipeConnectionTimeout = 1 * time.Minute
+
+	//NamedPipePathPrefix Represents prefix for named pipe path.
+	NamedPipePathPrefix = `\\.\pipe\ProtectedPrefix\Administrators\aws-vpc-k8s-connector`
+)
+
+// pipeOperator Represents interface with all operations on named piped.
+type pipeOperator interface {
+	start(ctx context.Context) error               // Start listener and accept connections on named pipe.
+	getOutput(ctx context.Context) (string, error) // Get output from named pipe.
+	getPath() string                               // Get named pipe path.
+}
+
+// pipeReadResult Represents result received from named pipe.
+type pipeReadResult struct {
+	output string // Represents message with output data.
+	error  error  // Represents error while getting output.
+}
+
+// pipe Represents named pipe.
+type pipe struct {
+	path       string              // Represents named pipe path.
+	listener   net.Listener        // Represents listener on named pipe.
+	resultChan chan pipeReadResult // Represents results chan to receive output from go routine.
+}
+
+// newPipe Create pipe object with random named pipe path.
+func newPipe() (pipeOperator, error) {
+	// Generate GUID random string.
+	g, err := windows.GenerateGUID()
+	if err != nil {
+		return nil, fmt.Errorf("error creating unique name for named pipe: %w", err)
+	}
+	// Create pipe path with prefix.
+	pipeName := fmt.Sprintf("%s-%s", NamedPipePathPrefix, g.String())
+	return &pipe{path: pipeName}, nil
+}
+
+// Start listener on named pipe. Start accepting connections on named pipe in separate go routine.
+// Creates result chan to get output from separate go routine.
+// Execute once per pipe.
+func (p *pipe) start(ctx context.Context) error {
```

Comment:
Please move these to a single lifetime function. You do not need this pattern of new, start, get output. Just make this a func called `getOutputFromPipe` and handle the case inline as the code can run synchronous on a single goroutine

Reply:
done